synapse/event_auth.py

```diff
@@ -83,6 +89,14 @@ def check(event, auth_events, do_sig_check=True, do_size_check=True):
         # FIXME
         return True
 
+    expected_depth = (max(event.prev_events, lambda x: x.depth) + 1)
```

Comment:
doesn't the `key` function have to be passed as a named argument?

```
max(event.prev_events, key=lambda x: x.depth)
```

However, the bigger problem here is that `event.prev_events` is a list of event ids, not a list of events. In the general case, we don't necessarily *have* the prev_events.

Reply:
Quite right. I think I glanced below and guessed it might have contained the events but that never seemed likey. And yeah, unless you defer this event until you have fetched the preceeding events and so on, then it could become quite stuck. Not to mention if there is some circular dependency maliciously ingested or if the preceeding event is invalid or has vanished..

Yet still we need some way to determine the most likey previous depth